fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/indexes/VersionIndexTest.java

```diff
@@ -2354,4 +2429,37 @@ private <M extends Message> void validateUsingNewerVersionFormat(@Nonnull List<F
         }
         assertFalse(versionKeyPairs.hasNext());
     }
+
+    @Nonnull
+    private List<Tuple> scanIndexToKeys(final RecordQueryPlannerConfiguration.IndexFetchMethod fetchMethod, final String indexName, final ScanProperties direction) throws Exception {
+        if (fetchMethod == RecordQueryPlannerConfiguration.IndexFetchMethod.SCAN_AND_FETCH) {
+            return recordStore.scanIndex(metaData.getIndex(indexName), IndexScanType.BY_VALUE, TupleRange.ALL, null, direction)
+                    .map(IndexEntry::getKey)
+                    .asList().get();
+        } else {
+            IndexScanBounds scanBounds = new IndexScanRange(IndexScanType.BY_VALUE, TupleRange.ALL);
+            KeyExpression primaryKey = recordStore.getRecordMetaData().getRecordType("MySimpleRecord").getPrimaryKey();
+            return recordStore.scanIndexRemoteFetch(indexName, scanBounds, primaryKey, null, direction, IndexOrphanBehavior.ERROR)
+                    .map(FDBIndexedRecord::getIndexEntry)
+                    .map(IndexEntry::getKey)
+                    .asList().get();
+        }
+    }
+
+    @Nonnull
+    private List<FDBIndexedRecord<Message>> scanIndexToRecords(final String indexName, final ScanProperties direction) throws Exception {
```

Comment:
I missed initially that this method only uses remote fetch. Is there a reason this doesn't take the `IndexFetchMethod` as a parameter?